delft3dcontainermanager/test_tasks.py

```diff
@@ -41,24 +44,42 @@ def test_do_docker_create(self):
         delay = do_docker_create.delay("image")
         self.assertEqual(delay.result, None)
 
-    def test_do_docker_start(self):
+    @patch('delft3dcontainermanager.tasks.Client', **mock_options)
+    def test_do_docker_start(self, mockClient):
         """
-        TODO: write test
+        Assert that the docker_start task
+        calls the docker client.start() function
         """
-        delay = do_docker_start.delay("id")
-        self.assertEqual(delay.result, False)
+        do_docker_start.delay("id")
+        mockClient.return_value.start.assert_called_with(container="id")
 
-    def test_do_docker_stop(self):
+    @patch('delft3dcontainermanager.tasks.Client', **mock_options)
+    def test_do_docker_stop(self, mockClient):
         """
-        TODO: write test
+        Assert that the docker_stop task
+        calls the docker client.stop() function
         """
-        delay = do_docker_stop.delay("id")
-        self.assertEqual(delay.result, False)
+        do_docker_stop.delay("id", timeout=5)
+        mockClient.return_value.stop.assert_called_with(
+            container="id", timeout=5)
 
-    def test_do_docker_remove(self):
+    @patch('delft3dcontainermanager.tasks.Client', **mock_options)
+    def test_do_docker_remove(self, mockClient):
         """
-        TODO: write test
+        Assert that the docker_remove task
+        calls the docker client.remove_container() function
         """
+        # Non running container
+        mockClient.return_value.inspect_container.return_value = {
+            "State": {"Running": False}}
+        delay = do_docker_remove.delay("id")
+        mockClient.return_value.remove_container.assert_called_with(
+            container="id")
+        self.assertEqual(delay.result, True)
+
+        # Running container
+        mockClient.return_value.inspect_container.return_value = {
+            "State": {"Running": True}}
         delay = do_docker_remove.delay("id")
         self.assertEqual(delay.result, False)
```

Comment:
Here also a check that the method is called zero times (we do not want to have accidental removals when the container is still running).